File: AngularApp/src/app/models/ProdutoServico.ts
import { Categoria } from "./Categoria";
import { Lancamento } from "./Lancamento";

export interface ProdutoServico {
    id?: number;
    descricao: string;
    preco: number;
    imagem: Blob | string;
    categoriaId: number,
    categoria: Categoria;
    lancamentos: Lancamento[];
}

File: ProjetoAspNetCoreApp/ClienteProduto.Domain/Entities/ProdutoServico.cs
﻿using System.Text.Json.Serialization;

namespace ClienteProjeto.Domain.Entities;

public class ProdutoServico : Entity
{
    public ProdutoServico(int id, string descricao)
    {
        Id = id;
        Descricao = descricao;
    }

    public string Descricao { get; set; }
    public double Preco { get; set; }
    public byte[] Imagem { get; set; }
    public int CategoriaId { get; set; }

    public Categoria Categoria { get; set; }

    [JsonIgnore]
    public ICollection<Lancamento> Lancamentos { get; set; }
    public ICollection<Pedido> Pedidos { get; set; }
}
